[PORT] Replace natural beheading with cranial fissures (splitting your skull)

Absolucy opened the pull request:
## About The Pull Request

Ports the following PRs:
- https://github.com/tgstation/tgstation/pull/80703
- https://github.com/tgstation/tgstation/pull/80935
- https://github.com/tgstation/tgstation/pull/84485
- https://github.com/tgstation/tgstation/pull/84822
- https://github.com/tgstation/tgstation/pull/88081


> Replaces natural beheading (doing a lot of damage to the head) with cranial fissures, which split your skull in half.
> 
> ![dreamseeker_2023-12-31T19-38-16](https://github.com/tgstation/tgstation/assets/35135081/f84dc479-6156-45b7-bb23-4e4ec7378f6b)
> 
> While you have this wound, your eyes can be pulled out of your head with bare hands, and slipping will spill your brain out of your head.
> 
> ~~Removes beheading objective from traitor.~~ (not applicable here)
> 
> Zombies can be beheaded all the same.
> 
> Any other way of beheading, such as surgery or amputation shears, is still possible.

## Why It's Good For The Game

Accidental beheadings because everyone just targets the damn head by default are way too common.

If you _really_ wanna behead someone, there's still ways to do so - spite is the most powerful motivator, after all.

> I've enjoyed seeing the results of the test merge to remove natural beheading. At 947 beheadings in a week, 1 in every 7 deaths resulted in a beheading. This makes it significantly easier to remove people from rounds as the brain is generally critical to actually reviving someone. While round removal is fine, it should be something that is intentionally performed with enough effort. Mass round removals ought to require a significant amount of effort. There are plenty of ways to round remove someone, but it ought not be an incidental choice.
> 
> The effects of beheading removal have shown some interesting anecdotal highlights (nothing here is backed up with stats, so take it with a grain of salt):
> 1. As a whole, people did not really replace it with anything. The biggest alternative has been lighting people on fire, but this has not been done at a significant enough scale at all to be a noteworthy problem, and is still fixable with enough effort.
> 2. Cult and rev rounds have played out far more interestingly. In one round I was adminning a head of staff who intended to behead every revolutionary they saw, but because they didn't, the back and forth continues. I've also had opportunities to revive head revs that I am skeptical would've been available to me otherwise.
> 
> Complete removal of beheading was not chosen as it does not feel right for a repeated fire axe to the head to do basically nothing. The current implementation is intended to be something that is just not useful enough to do on everyone you see, but is still appreciably an effect.

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl: Absolucy, Mothblocks
balance: Instead of too much damage to the head beheading someone, it will now split their skull in half. While their skull is open, you can rip out their eyes with your hands. and they will spill their brain out of their head if they slip. Zombies can still be beheaded, though.
balance: The Path of Blades ascension will accept either a beheaded person, or someone with their skull split open.
add: (nikothedude) Being cursed now enables disembowlements/cranial fissures outside of hardcrit.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Absolucy commented:
@Ice-Type The main thing is that I believe that beheading should require deliberate, intentional action

Almost everyone targets the head by default, likely due to head armor being slightly less common, so accidental beheadings are actually quite common with anyone using a sharp weapon.

I do have an idea, tho: you can use a sufficiently sharp weapon on a dead body with a cranial fissure, while targeting the head, to decapitate them after a delay.

GhostEcho commented:
Theres been a lot of seemingly unintended beheading and deorganing tbh, but anecdotal

Ice-Type commented:
> @Ice-Type The main thing is that I believe that beheading should require deliberate, intentional action. It shouldn't really be something you do by _accident_.
> 
> Almost everyone targets the head by default, likely due to head armor being slightly less common, so accidental beheadings are actually quite common with anyone using a sharp weapon.
> 
> I do have an idea, tho: what if you could use a sufficiently sharp weapon on a dead body with a cranial fissure, while targeting the head, to decapitate them after a delay?

This would be a pretty good compromise - though I do have a few suggestions
1) Gutting ends up being pretty similar to a decap in terms of damage done. Maybe that should be similarly deliberate
2) Would it be possible to make the progress bar scale with how well an item wounds?
3) This still leaves bullets in a bit of a spot, could at least projectiles with dismemberment (the sniper rifle, mainly, also the syndie xbow) ignore this and just straight decap?

Also slasher needs taking a look at since decaps are how you kill it, and the zombie specific decap mechanic should be removed if you do add the decap progress bar.

GhostEcho commented:
This potentially may not allow slashers to get decapped

MissClassyPineapple commented:
We have found an issue with organ manipulation surgeries done on cursed oozelings. Because they are cursed, they are disembowelled as soon as the surgery starts, then because their core is in their chest and leaving their body via disembowelment, they instantly die.

Coll6 commented:
> We have found an issue with organ manipulation surgeries done on cursed oozelings. Because they are cursed, they are disembowelled as soon as the surgery starts, then because their core is in their chest and leaving their body via disembowelment, they instantly die.

Ooh yeah okay I can look into that.

Absolucy commented:
> We have found an issue with organ manipulation surgeries done on cursed oozelings. Because they are cursed, they are disembowelled as soon as the surgery starts, then because their core is in their chest and leaving their body via disembowelment, they instantly die.

@MissClassyPineapple fixed

Absolucy commented:
I've made it so you can still decap someone with a weapon, however that can only happen if their head is at max damage (duh), they already have a cranial fissure, and have been dead for at least 15 seconds.

15 seconds feels like it's a good enough threshold to where if you continue slashing at someone's head for that long _after_ they're dead, you prolly _WANT_ their head

https://github.com/user-attachments/assets/95ed7f39-d5e5-4605-a0f8-f97ee0df10bb

Ice-Type commented:
> I've made it so you can still decap someone with a weapon, however that can only happen if their head is at max damage (duh), they already have a cranial fissure, and have been dead for at least 15 seconds.
> 
> 15 seconds feels like it's a good enough threshold to where if you continue slashing at someone's head for that long _after_ they're dead, you prolly _WANT_ their head
> who.you.looking.at.denthead.mp4

Could be shorter really. Five seconds?

Ice-Type commented:
But really while it's good for functionality purposes I think there's a more elegant way to implement it. A progress bar would be ideal.

Absolucy commented:
alright i've changed the "you can decap 15+ seconds after death" thing - instead, it's a progress bar now. right click on a dead body while targeting the head to cut their head off

More powerful weapons do this quicker. Note that a weapon needs to be both sharp and have a force of _at least 15_ to be used to be head this way:
- 15-24 force: 10 seconds
- 25-30 force: 7.5 seconds
- 31+ force: 5 seconds

(note, this clip was recorded before I added sounds and made it so stronger weapons are quicker)

https://github.com/user-attachments/assets/739824eb-1923-4b30-bbde-aba1c02904dc

Ice-Type commented:
> alright i've changed the "you can decap 15+ seconds after death" thing - instead, it's a progress bar now. right click on a dead body while targeting the head to cut their head off
> 
> More powerful weapons do this quicker. Note that a weapon needs to be both sharp and have a force of _at least 15_ to be used to be head this way:
> 
>     * 15-24 force: 10 seconds
> 
>     * 25-30 force: 7.5 seconds
> 
>     * 31+ force: 5 seconds
> 
> 
> (note, this clip was recorded before I added sounds and made it so stronger weapons are quicker)
> behead.mp4

I think this is pretty great.

Tostah commented:
> I feel like this loses a lot of flavour, and that we don't really need this. Being able to just... chop someone's head off with a lightsabre or an axe feels right, and so the sandbox gains a lot from said thing working.

This, essentially. It just feels right to be able to lob someone's head off, even accidentally. 

I think the real problem here is how beheading is seen as round removal. A better solution would be making placing heads back on bodies easier through non-surgical means.

Blacklist897 commented:
Cutting someones head off mid fight was extremely funny, this pr was crap on tg, and near universally disliked, I dont think it should be here

Absolucy commented:
@Blacklist897 
> Cutting someones head off mid fight was extremely funny
not for the other player...

> this pr was crap on tg, and near universally disliked, I dont think it should be here
the difference is that **you can still decap using weapons** with this version, it just requires you to intentionally do so

i've made it so decapping continues to play attack animations+sounds while you're doing it, to hopefully better satisfy the more bloodthirsty players

Absolucy commented:
@ThePooba should i make slashers beheadable tho, or maybe just take longer to get back up if they have a fissure?

ThePooba commented:
> @ThePooba should i make slashers beheadable, or maybe just take longer to get back up if they have a fissure?

mmm make it take longer to get back up i think
